packages/immutadot/src/path/toPath.js

```diff
@@ -117,14 +118,28 @@ const sliceNotationParser = map(
   ([sliceStart, sliceEnd, rest]) => [[slice, [toSliceIndex(sliceStart), toSliceIndex(sliceEnd)]], ...stringToPath(rest)],
 )
 
-const pathSegmentEndedByDotParser = map(
-  regexp(/^([^.[]*?)\.(.*)$/),
-  ([beforeDot, afterDot]) => [[prop, beforeDot], ...stringToPath(afterDot)],
+const listPropRegexp = /^,?((?!["'])([^,]*)|(["'])(.*?[^\\])\3)(.*)/
+const listNotationParser = map(
+  regexp(/^\{(((?!["'])[^,}]*|(["']).*?[^\\]\2)(,((?!["'])[^,}]*|(["']).*?[^\\]\6))*)\}\.?(.*)$/),
+  ([rawProps, , , , , , rest]) => {
+    const props = []
+    if (rawProps.startsWith(',')) props.push('')
+    for (let propMatch = rawProps.match(listPropRegexp); propMatch !== undefined; propMatch = propMatch[5] === '' ? undefined : propMatch[5].match(listPropRegexp))
+      props.push(propMatch[2] === undefined ? propMatch[4] : propMatch[2])
+    return props.length === 1 ? [[prop, props[0]], ...stringToPath(rest)] : [[list, props], ...stringToPath(rest)]
+  },
+)
+
+const incompleteListNotationParser = map(
+  regexp(/^(\{[^.[{]*)(.*)$/),
+  ([beforeNewSegment, rest]) => {
+    return [[prop, beforeNewSegment], ...stringToPath(rest)]
+  },
```

Comment:
Maybe the lambda could be written in the expression form (ie without braces and `return`)?